jwst/associations/lib/rules_level2_base.py

```diff
@@ -1027,3 +1030,147 @@ def _init_hook(self, item):
 
         super(AsnMixin_Lv2Spectral, self)._init_hook(item)
         self.data['asn_type'] = 'spec2'
+
+
+class AsnMixin_Lv2WFSS:
+    """Utility and overrides for WFSS associations"""
+
+    def add_catalog_members(self):
+        """Add catalog and direct image member based on direct image members"""
+        directs = self.members_by_type('direct_image')
+        if not directs:
+            raise AssociationNotValidError(
+                '{} has no required direct image exposures'.format(
+                    self.__class__.__name__
+                )
+            )
+
+        sciences = self.members_by_type('science')
+        if not sciences:
+            raise AssociationNotValidError(
+                '{} has no required science exposure'.format(
+                    self.__class__.__name__
+                )
+            )
+        science = sciences[0]
+
+        # Get the exposure sequence for the science. Then, find
+        # the direct image greater than but closest to this value.
```

Comment:
What exactly does "direct image greater than" translate to? Does that refer to the chronological order in which the exposures were obtained and hence suggests that the chosen direct image must be taken *after* the grism/science exposure? I'm not sure there's any enforced ordering of the direct and grism exposures in a WFSS observation.

Reply:
The "order" referred to here is the "EXPSPCIN" order. It is the order defined by APT. This is seen in the bottom of either WFSS template under "Exposure Sequence Display". Direct images, when requested, are always after the science that requested it, where the "Direct Image" checkbox is set in the above "Science (Grism) Exposures".

I do not know if this enforces sequencing on the observatory itself, but it does define the EXPSPCIN values. So, one wants to find the nearest direct image that has an EXPSPCIN value greater than the current science one. You actually fixed a bug in this last year; the logic was reversed?